nameserver/mocks_test.go

```diff
@@ -436,6 +437,7 @@ func (clk *mockedClock) Forward(secs int) {
 	Debug.Printf(">>>>>>> Moving clock forward %d seconds - Time traveling >>>>>>>", secs)
 	clk.Add(time.Duration(secs) * time.Second)
 	Debug.Printf("<<<<<<< Time travel finished! We are at %s <<<<<<<", clk.Now())
+	runtime.Gosched() // give some time to other goroutines
```

Comment:
I see you commented in the original bug about this - it'd be good to import some of that rationale here in the form of an explicatory comment